checkov/kustomize/runner.py

```diff
@@ -118,22 +125,88 @@ def mutate_kubernetes_results(
                 # Fix file path to repo relative path
                 if self.original_root_dir:
                     repo_dir = str(pathlib.Path(self.original_root_dir).resolve())
+
+                    if env_vars_config.ALLOW_KUSTOMIZE_FILE_EDITS:
+                        caller_file_line_range, caller_file_path = self._get_caller_file_info(entity_context, k8_file,
+                                                                                              k8_file_path, resource_id,
+                                                                                              root_folder)
+
                     if realKustomizeEnvMetadata['filePath'].startswith(repo_dir):
                         file_path = realKustomizeEnvMetadata['filePath'][len(repo_dir):]
 
             code_lines = entity_context.get("code_lines")
             file_line_range = self.line_range(code_lines)
+
             record = Record(
                 check_id=check.id, bc_check_id=check.bc_id, check_name=check.name,
-                check_result=check_result, code_block=code_lines, file_path=file_path,
-                file_line_range=file_line_range,
+                check_result=check_result, code_block=code_lines,
+                file_path=file_path, file_line_range=file_line_range,
+                caller_file_path=caller_file_path, caller_file_line_range=caller_file_line_range,
                 resource=kustomizeResourceID, evaluations=variable_evaluations,
                 check_class=check.__class__.__module__, file_abs_path=absolute_file_path, severity=check.severity)
             record.set_guideline(check.guideline)
             report.add_record(record=record)
 
         return report
 
+    def _get_caller_file_info(self, entity_context: _EntityContext, k8_file: str, k8_file_path: str, resource_id: str,
+                              root_folder: str | None) -> tuple[tuple[int, int] | None, str]:
+        origin_relative_path = entity_context['origin_relative_path']
+        k8s_file_dir = pathlib.Path(k8_file_path).parent
+        raw_file_path = k8s_file_dir / origin_relative_path
+        caller_file_path = self._get_caller_file_path(k8s_file_dir, origin_relative_path, raw_file_path)
+        if root_folder is None:
+            return None, caller_file_path
+        caller_file_line_range = self._get_caller_line_range(root_folder, k8_file, origin_relative_path,
+                                                             resource_id)
+        return caller_file_line_range, caller_file_path
+
+    @staticmethod
+    def _get_caller_file_path(k8s_file_dir: pathlib.Path, origin_relative_path: str, raw_file_path: pathlib.Path)\
+            -> str:
+        amount_of_parents = str.count(origin_relative_path, '..')
+        directory_prefix_path = k8s_file_dir
+        for _ in range(amount_of_parents):
+            directory_prefix_path = directory_prefix_path.parent
```

Comment:
```suggestion
        directory_prefix_path = k8s_file_dir
        if amount_of_parents:
            directory_prefix_path = k8s_file_dir.parents[amount_of_parents - 1]
```
I think this should do the same, without looping 🙂